java-util/src/main/java/io/druid/java/util/common/parsers/TimestampParser.java

```diff
@@ -47,13 +48,16 @@
         public DateTime apply(String input)
         {
           Preconditions.checkArgument(input != null && !input.isEmpty(), "null timestamp");
+          String inputWithoutTimeZone = ParserUtils.removeTimeZone(input);
+          DateTimeZone timeZone = ParserUtils.extractTimeZone(input);
+
           for (int i = 0; i < input.length(); i++) {
```

Comment:
I think the above two methods are reasonable to call only inside this loop, inside the condition `(input.charAt(i) < '0' || input.charAt(i) > '9')`. Because I don't think we expect input like `1234567890 UTC`. Also those methods could be joined together and optimized by checking counts of `' '` (space) occurrences and using `indexOf()` and `substring()`, rather than unconditionally splitting using regex string, that incurs creation of `Pattern` and `Matcher` objects, and then string concatenation again.

Reply:
@leventov Made the suggested changes